Update M2-emacs for version 1.25.05

d-torrance opened the pull request.

mikestillman commented:
@d-torrance what is the best way for me to test this, without changing my M2-emacs yet?

d-torrance commented:
The indentation/electric braces changes haven't gone in yet, since there's still too many issues.  The only change since the last release is a hacky solution to syntax highlighting of unterminated strings and comments.  This PR is just basic stuff like updating the version number, documentation, and adding some new symbols for syntax highlighting and auto-completion.

Probably the simplest way to try out the new code would be to download https://raw.githubusercontent.com/Macaulay2/M2-emacs/refs/heads/main/M2.el, open it in a buffer, click on the "Emacs-Lisp" menu and select "Evaluate Buffer".  Then fire up a new M2 comint buffer and try:

```
print "\""
print "-*"
```

Prior to the recent changes, the syntax highlighting would have been messed up, but now it shouldn't be.  (It's super hacky -- it prints an invisible closing `"` and `*-` before the next prompt, but it works!)

mahrud commented:
> I'll try using it while working on my PR's and issues the next 2 days...

Isn't the plan for a release tomorrow? That would mean building the distribution packages tonight or tomorrow morning at most. Which PRs do you still want included?
